crates/re_renderer/examples/multiview.rs

```diff
@@ -177,6 +164,83 @@ fn random_color(rnd: &mut impl rand::Rng) -> Color32 {
     .into()
 }
 
+impl Multiview {
+    fn draw_view<D: 'static + re_renderer::renderer::DrawData + Sync + Send + Clone>(
+        &mut self,
+        re_ctx: &mut RenderContext,
+        target_cfg: TargetConfiguration,
+        skybox: &GenericSkyboxDrawData,
+        draw_data: &D,
+        index: u32,
+    ) -> (ViewBuilder, wgpu::CommandBuffer) {
+        let mut view_builder = ViewBuilder::default();
+        view_builder.setup_view(re_ctx, target_cfg).unwrap();
+
+        if self
+            .take_screenshot_next_frame_for_view
+            .map_or(false, |i| i == index)
+        {
+            self.scheduled_screenshots
+                .push(view_builder.schedule_screenshot(re_ctx).unwrap());
+            re_log::info!("Scheduled screenshot for view {}", index);
+        }
+
+        let command_buffer = view_builder
+            .queue_draw(skybox)
+            .queue_draw(draw_data)
+            .draw(re_ctx, Rgba::TRANSPARENT)
+            .unwrap();
+
+        (view_builder, command_buffer)
+    }
+
+    fn handle_incoming_screenshots(&mut self, re_ctx: &mut RenderContext) {
+        re_ctx
+            .gpu_readback_belt
+            .lock()
+            .receive_data(|data, identifier| {
+                if let Some(index) = self
+                    .scheduled_screenshots
+                    .iter()
+                    .position(|s| s.identifier == identifier)
+                {
+                    let screenshot = self.scheduled_screenshots.swap_remove(index);
+
+                    // Need to do a memcpy to remove the padding.
+                    re_log::info!("Received screenshot. Total bytes {:?}", data.len());
+                    let row_info = screenshot.row_info;
+                    let mut buffer = Vec::with_capacity(
+                        (row_info.bytes_per_row_unpadded * screenshot.height) as usize,
+                    );
+                    for row in 0..screenshot.height {
+                        let offset = (row_info.bytes_per_row_padded * row) as usize;
+                        buffer.extend_from_slice(
+                            &data[offset..(offset + row_info.bytes_per_row_unpadded as usize)],
+                        );
+                    }
+
+                    // Get next available file name.
+                    let mut i = 0;
+                    let mut filename = "screenshot.png".to_owned();
+                    while std::path::Path::new(&filename).exists() {
+                        filename = format!("screenshot_{i}.png");
+                        i += 1;
+                    }
```

Comment:
the fact that it doesn't start at `screenshot_0.png` is pretty annoying :grimacing: